services/common/src/utils/helpers.ts

```diff
@@ -83,16 +84,29 @@ export const formatPermitConditionStep = (step: string) => {
   return "";
 }
 
-export const getConditionsWithRequirements = (conditions: IPermitCondition[]) => {
+export const getConditionsWithRequirements = (conditions: IPermitCondition[], requirements?: IMineReportPermitRequirement[]) => {
   let result = [];
+
+  const requirementsByCondition = requirements?.length ? requirements.reduce((acc, requirement) => {
+    if (!acc[requirement.permit_condition_id]) {
+      acc[requirement.permit_condition_id] = [];
+    }
+    acc[requirement.permit_condition_id].push(requirement);
+    return acc;
+  }, {}) : {};
+
   conditions.forEach((condition) => {
-    if (condition.mineReportPermitRequirement) {
+    if (requirements && condition?.permit_condition_id) {
+      const req = requirementsByCondition[condition.permit_condition_id];
+      result = [...result, ...(req || [])];
+    } else if (condition?.mineReportPermitRequirement) {
       result.push(condition);
     }
 
-    if (condition.sub_conditions && condition.sub_conditions.length > 0) {
+    if (condition?.sub_conditions && condition?.sub_conditions.length > 0) {
```

Comment:
just a guess- sonarcloud is probably telling you to simplify to condition?.sub_conditions?.length > 0